asapdiscovery-data/asapdiscovery/data/dask_utils.py

```diff
@@ -1,26 +1,46 @@
 import logging
 from collections.abc import Iterable
-from typing import Optional
+from pathlib import Path
+from typing import Any, Callable, Optional, Union
 
 import dask
+import psutil
 from asapdiscovery.data.enum import StringEnum
+from asapdiscovery.data.execution_utils import (
+    get_platform,
+    guess_network_interface,
+    hyperthreading_is_enabled,
+)
 from dask import config as cfg
 from dask.utils import parse_timedelta
 from dask_jobqueue import LSFCluster
 from distributed import Client, LocalCluster
 from pydantic import BaseModel, Field
 
-from .execution_utils import guess_network_interface
-
 logger = logging.getLogger(__name__)
 
 
+class BackendType(StringEnum):
+    """
+    Enum for backend types indicating how data is being passed into the dask function, either an in-memory object,
+    or a JSON file of that object on disk
+    """
+
+    IN_MEMORY = "in-memory"
+    DISK = "disk"
+
+
 def set_dask_config():
     cfg.set({"distributed.scheduler.worker-ttl": None})
     cfg.set({"distributed.admin.tick.limit": "4h"})
-    cfg.set({"distributed.scheduler.allowed-failures": 1})
+    cfg.set({"distributed.scheduler.allowed-failures": 2})
```

Comment:
Why the change to 2 here?

Reply:
Basically if the work is borked, leading to a killed worker and the loss of a lot of progress we don't try it 3 times, instead only 2. 1 is too few IMHO and 3 can be a bit slow.